datasync/transport.go

```diff
@@ -0,0 +1,61 @@
+package datasync
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"errors"
+	"github.com/golang/protobuf/proto"
+	datasyncpeer "github.com/status-im/status-protocol-go/datasync/peer"
+	"github.com/vacp2p/mvds/protobuf"
+	"github.com/vacp2p/mvds/state"
+	"github.com/vacp2p/mvds/transport"
+)
+
+var errNotInitialized = errors.New("Datasync transport not initialized")
+
+type DataSyncNodeTransport struct {
+	packets  chan transport.Packet
+	dispatch func(context.Context, *ecdsa.PublicKey, []byte) error
+}
+
+func NewDataSyncNodeTransport() *DataSyncNodeTransport {
+	return &DataSyncNodeTransport{
+		packets: make(chan transport.Packet),
+	}
+}
+
+func (t *DataSyncNodeTransport) Init(dispatch func(context.Context, *ecdsa.PublicKey, []byte) error) {
+	t.dispatch = dispatch
+}
+
+func (t *DataSyncNodeTransport) AddPacket(p transport.Packet) {
+	t.packets <- p
+}
+
+func (t *DataSyncNodeTransport) Watch() transport.Packet {
+	return <-t.packets
+}
+
+func (t *DataSyncNodeTransport) Send(_ state.PeerID, peer state.PeerID, payload protobuf.Payload) error {
```

Comment:
I don't understand why there are two state.PeerIDs and one is ignored

Reply:
that's the interface of `Send` in mvds, the first one is the sender, the other is the destination. If it needs documenting probably the best place is in `mvds` where the interface is defined